general/development/policies/codingstyle-moodleapp.md

```diff
@@ -465,3 +541,101 @@ export default class MyComponent {}
 ```
 
 </InvalidExample>
+
+### Declaring page modules
+
+When creating a page component, it should be declared in the feature's [lazy modules](../../../general/app/development/development-guide.md#routing). Exceptionally, pages that are used by more than one module can create a page module; but this module should only declare components, it shouldn't include any routing functionality.
+
+<ValidExample title="Good">
+
+```ts
+// file: core/features/feature/pages/index/index.ts
+@Component({
+    selector: 'page-core-feature-index',
+    templateUrl: 'index.html',
+})
+export class CoreFeatureIndexPageComponent {}
+```
+
+```ts
+// file: core/features/feature/feature-lazy.module.ts
+const routes: Routes = [
+    {
+        path: 'feature',
+        component: CoreFeatureIndexPageComponent,
+    },
+];
+
+@NgModule({
+    imports: [
+        RouterModule.forChild(routes),
+        CoreSharedModule,
+    ],
+    declarations: [
+        CoreFeatureIndexPageComponent,
+    ],
+})
+export class CoreFeatureLazyModule {}
+```
+
+</ValidExample>
+
+<CodeExample type="warning" title="Allowed only if the page is used in multiple modules">
+
+```ts
+// file: core/features/feature/pages/index/index.page.ts
+@Component({
+    selector: 'page-core-feature-index',
+    templateUrl: 'index.html',
+})
+export class CoreFeatureIndexPageComponent {}
+```
+
+```ts
+// file: core/features/feature/pages/index/index.module.ts
+@NgModule({
+    imports: [
+        CoreSharedModule,
+    ],
+    declarations: [
+        CoreFeatureIndexPageComponent,
+    ],
+})
+export class CoreFeatureLazyModule {}
```

Comment:
Maybe you should use a different module name to make it more clear that it's a "page module", not the whole feature's lazy module? Also in the next example.